exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -145,132 +168,250 @@ func (e *Exchange) SetDefaults() {
+	}); err != nil {
+		return err
+	}
+	wsFutures, err := e.API.Endpoints.GetURL(exchange.WebsocketFutures)
 	if err != nil {
 		return err
 	}
+	// Futures Public Connection
+	if err := e.Websocket.SetupNewConnection(&websocket.ConnectionSetup{
+		URL:                   wsFutures,
+		ResponseCheckTimeout:  exch.WebsocketResponseCheckTimeout,
+		RateLimit:             request.NewWeightedRateLimitByDuration(2 * time.Millisecond),
+		ResponseMaxLimit:      exch.WebsocketResponseMaxLimit,
+		Handler:               e.wsFuturesHandleData,
+		Subscriber:            e.SubscribeFutures,
+		Unsubscriber:          e.UnsubscribeFutures,
+		GenerateSubscriptions: e.generateFuturesSubscriptions,
+		Connector:             e.WsFuturesConnect,
+		MessageFilter:         connFuturesPublic,
+	}); err != nil {
+		return err
+	}
 
+	wsFuturesPrivate, err := e.API.Endpoints.GetURL(exchange.WebsocketFuturesPrivate)
+	if err != nil {
+		return err
+	}
+	// Futures Private Connection
 	return e.Websocket.SetupNewConnection(&websocket.ConnectionSetup{
-		ResponseCheckTimeout: exch.WebsocketResponseCheckTimeout,
-		ResponseMaxLimit:     exch.WebsocketResponseMaxLimit,
+		URL:                   wsFuturesPrivate,
+		ResponseCheckTimeout:  exch.WebsocketResponseCheckTimeout,
+		ResponseMaxLimit:      exch.WebsocketResponseMaxLimit,
+		RateLimit:             request.NewWeightedRateLimitByDuration(2 * time.Millisecond),
+		Handler:               e.wsFuturesHandleData,
+		Subscriber:            e.SubscribeFutures,
+		Unsubscriber:          e.UnsubscribeFutures,
+		GenerateSubscriptions: e.generateFuturesPrivateSubscriptions,
+		Connector:             e.futuresAuthConnect,
+		MessageFilter:         connFuturesPrivate,
+		Authenticate:          e.authenticateFuturesAuthConn,
+		Authenticated:         true,
 	})
 }
 
-// FetchTradablePairs returns a list of the exchanges tradable pairs
-func (e *Exchange) FetchTradablePairs(ctx context.Context, _ asset.Item) (currency.Pairs, error) {
-	// TODO: Upgrade to new API version for fetching operational pairs.
-	resp, err := e.GetTicker(ctx)
-	if err != nil {
-		return nil, err
-	}
+// FetchTradablePairs returns a list of the exchange's tradable pairs
+func (e *Exchange) FetchTradablePairs(ctx context.Context, assetType asset.Item) (currency.Pairs, error) {
+	switch assetType {
+	case asset.Spot, asset.Margin:
+		resp, err := e.GetSymbols(ctx)
+		if err != nil {
+			return nil, err
+		}
 
-	pairs := make([]currency.Pair, 0, len(resp))
-	for key, info := range resp {
-		// Poloniex returns 0 for highest bid and lowest ask if support has been
-		// dropped from the front end. We don't want to add these pairs.
-		if info.HighestBid == 0 || info.LowestAsk == 0 {
-			continue
+		pairs := make([]currency.Pair, 0, len(resp))
+		for _, symbolDetail := range resp {
+			if strings.EqualFold(symbolDetail.State, "PAUSE") {
+				continue
+			}
+			cp, err := currency.NewPairFromString(symbolDetail.Symbol)
```

Comment:
May as well make `symbolDetail.Symbol` be a `currency.Pair`, saving all these extra calls